### PR TITLE
Update UG (fix inaccurate commands, improve ordering)

### LINKED ISSUE #130: [PE-D][Tester C] add team feature does not work as specified in user guide
The format and examples for the command `add team` is specified in the user guide as follows below.

However, in the app, when following the example command given, an error message is given

Upon following the format given in the `--help` command, I realised that the `-n` prefix is not needed and the command works once it is removed.

This might be confusing to users who follow the commands based on the user guide, especially if it is their first time using the app.

-------------
Labels: `severity.Medium` `type.DocumentationBug`
original: normkoh/ped#1

### LINKED ISSUE #132: [PE-D][Tester A] Unable to use command : Creating a new team add team command
Problems: Examples stated below given in the user guide is not working. The `add team --help` message is difficult to understand and read.
- `add team -n CS2103`
- `add team -n CS2102 -d "Database Systems"`

Have to try and test for 5 minutes to figure out how to use the command.

-------------
Labels: `severity.High` `type.DocumentationBug`
original: LianGuoYang/ped#2

### LINKED ISSUE #135: [PE-D][Tester D] Executing "add team -n CS2103" does not add team
Expected a new team to be added based on the user guide, but received an invalid command format message instead. 

-------------
Labels: `severity.VeryLow` `type.DocumentationBug`
original: pewggls/ped#1

### LINKED ISSUE #137: [PE-D][Tester B] Edit deadline doesn't actually work as documented
According to User Guide, edit task deadline should be:

but this returns an error. Instead,

`edit task 1 -d 2022-10-10 23:59` without quotations works

-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: gunbux/ped#4

### LINKED ISSUE #141: [PE-D][Tester A] Command not working as expected: Add task to team : add task
Problem: User guide shown below. However when I want to add a new task with only the name, its not working.

Solution: Clarify how to add a new task with the name only.

-------------
Labels: `severity.Medium` `type.DocumentationBug`
original: LianGuoYang/ped#5

### LINKED ISSUE #142: [PE-D][Tester A] Command not working : Add task to team : add task
Problem: Examples shown in User Guide not working. 

- `add task -n "Merge PR" -a 1 -d "2022-12-02 23:59"`
- `add task -n "Review PR" -a 1 3 -d "2022-12-02 23:59"`

-------------
Labels: `severity.Medium` `type.DocumentationBug`
original: LianGuoYang/ped#6

### LINKED ISSUE #144: [PE-D][Tester A] Add task error message
Problem: Possible wrong error description shown for add task. Is the datetime incorrect?

-------------
Labels: `type.FeatureFlaw` `severity.Low`
original: LianGuoYang/ped#8

### LINKED ISSUE #146: [PE-D][Tester A] Edit task error message
Problem: Possible wrong error description shown for edit task. Is the datetime incorrect?

-------------
Labels: `type.FeatureFlaw` `severity.Low`
original: LianGuoYang/ped#9

### LINKED ISSUE #147: [PE-D][Tester C] Unable to add task even with valid date time format
The format and example for the `add task` feature is given in the user guide as follows:

When copying and entering the command given in the first example, an error message is shown about the dateTime having the wrong format

Trying the dateTime given in the error message does not work either

I tried other command formats, but they do not work either

Users will be unable to add tasks with deadlines, and so this affects the functionality of the `add task` feature

-------------
Labels: `severity.High` `type.FunctionalityBug`
original: normkoh/ped#4

### LINKED ISSUE #148: [PE-D][Tester A] Command shown in UG not working: Set Deadline for task: set deadline
Problem: Example shown in UG not working.

-------------
Labels: `severity.Medium` `type.DocumentationBug`
original: LianGuoYang/ped#10

### LINKED ISSUE #149: [PE-D][Tester A] Set deadline error message
Problem: Is there any error with the datetime shown? 

-------------
Labels: `type.FeatureFlaw` `severity.Medium`
original: LianGuoYang/ped#11

### LINKED ISSUE #151: [PE-D][Tester C] add task feature does not work with format specified in user guide
The format for the `add task` feature is specified as follows in the user guide:

I tried to add task with only the name parameter given, but an error message was given 

Removing the `-n` prefix solved the problem and task was added.

This might be confusing to users using the application for the first time and are following the user guide.

-------------
Labels: `severity.Medium` `type.DocumentationBug`
original: normkoh/ped#5

### LINKED ISSUE #152: [PE-D][Tester A] Sort members : sort member not working
Problem: Unable to understand how to use this function.

-------------
Labels: `severity.Medium` `type.DocumentationBug`
original: LianGuoYang/ped#13

### LINKED ISSUE #153: [PE-D][Tester A] Unable to use command : Sort tasks : sort task
Problem: Do not know how to use this command.

-------------
Labels: `severity.Medium` `type.DocumentationBug`
original: LianGuoYang/ped#14

### LINKED ISSUE #157: [PE-D][Tester D] Add task format in User Guide is invalid
Expected the command add task -n "Merge PR" to add a task, but gave an error message "Invalid command format! Try 'add task --help' for more information." instead.

-------------
Labels: `severity.Low` `type.DocumentationBug`
original: pewggls/ped#5

### LINKED ISSUE #158: [PE-D][Tester C] Keywords for sort are not explained
The format and examples for the `sort` command are as specified in the user guide:

However, there is no mention of what the terms `asc` `dsc` and `res` means.

This might be confusing to users who have to figure out what each term means by testing it out themselves.

-------------
Labels: `severity.Medium` `type.DocumentationBug`
original: normkoh/ped#6

### LINKED ISSUE #161: [PE-D][Tester D] Invalid DateTime format error message shown even though format is correct
For example, the command add task "task1" -d "2022-01-01 10:01" gives the message "Invalid command format! DateTime should have the following format yyyy-MM-dd HH:mm (e.g. 2022-01-01 10:01)" even though the format should be correct.

-------------
Labels: `type.FunctionalityBug` `severity.Medium`
original: pewggls/ped#6

### LINKED ISSUE #163: [PE-D][Tester A] Unmark command in UG
Problem: After I accidentally `mark` the wrong tasks, how do I `unmark` the task? Unable to find `unmark` command in UG.

-------------
Labels: `severity.Medium` `type.DocumentationBug`
original: LianGuoYang/ped#19

### LINKED ISSUE #168: [PE-D][Tester D] User guide typo in "sort" commands
The "sort task" and "sort member" commands should be "sort tasks" and "sort members" instead

-------------
Labels: `severity.VeryLow` `type.DocumentationBug`
original: pewggls/ped#10

### PR DESCRIPTION
**Pull request by mjgui**
Just an initial PR to improve the UG

## Other potential things to do

- [ ] Update the command summary table (and potentially move it to the top)
- [ ] Add hyperlinks to navigate around the document
- [ ] Improve the language at the top of the User Guide
- [ ] Improve based on CS2101 feedback

## Fixes

Fixes #168
Fixes #163
Fixes #161
Fixes #158
Fixes #157
Fixes #153
Fixes #152
Fixes #151
Fixes #149
Fixes #148
Fixes #147
Fixes #146
Fixes #144
Fixes #142
Fixes #141
Fixes #137
Fixes #135
Fixes #130

Partially fixes but does not close #132 (confusing error messages)

### CODE REVIEW

**Comment by marcuspang** [comment]
Fixes #168

File: docs/UserGuide.md [hunk 1]
```diff
@@ -223,36 +341,92 @@ Examples:
...
+
+Examples
+* `find task -n User Guide` finds tasks with **names** containing **either** the word "User" or "Guide".
+
 ### List tasks in team: `list tasks`
 
 View all the tasks currently in the user’s team in the form of a list. List can also be filtered based on complete or
 remaining tasks.
 
+The command `list tasks` can also be used to view all tasks again after the `find task` command has been run.
```
Comment (Devanshshah1309) [inline docs/UserGuide.md:393]: "... since the `find task` command will filter the tasks by some keyword(s)"

File: docs/UserGuide.md [hunk 2]
```diff
@@ -223,36 +341,92 @@ Examples:
...
-Mark a specified task as done.
+View the number of tasks assigned to each user in the team.
 
 Format:
 
-- `mark INDEX`
+- `summary`
 
-Examples:
+### Sort tasks : `sort tasks`
```
Comment (Devanshshah1309) [inline docs/UserGuide.md:409]: Can I check why sort has `tasks` but find has `task`? We should probably standardise everything to use `task`?

File: docs/UserGuide.md [hunk 3]
```diff
@@ -223,36 +341,92 @@ Examples:
...
 
-### View summary of task assignments in team: `summary`
+Examples
+* `sort tasks asc` sorts the tasks in **ascending** order.
+* `sort tasks dsc` sorts the tasks in **descending** order.
+* `sort tasks res` **resets** the order of the tasks shown.
 
-View the number of tasks assigned to each user in the team.
+### Filter tasks by team member : `tasksof`
+Find all tasks in the current team assigned to a particular team member.
```
Comment (Devanshshah1309) [inline docs/UserGuide.md:421]:
```suggestion
Find all tasks in your current team that have been assigned to a particular team member.
```

File: docs/UserGuide.md [hunk 4]
```diff
@@ -223,36 +341,92 @@ Examples:
...
 
 Format:
+* `tasksof MEMBER_INDEX`
 
-- `summary`
+Examples
+* `tasksof 1` will show all tasks assigned to the first member in the team's member list.
```
Comment (Devanshshah1309) [inline docs/UserGuide.md:427]: We might want to look through the formatting of our examples again to make sure its standardised.
```suggestion
* `tasksof 1` will show all tasks assigned to the first member in your current team's member list.
```

File: docs/UserGuide.md [hunk 5]
```diff
@@ -264,7 +438,7 @@ Format:
 
 Examples:
 
-- `add link -n google -l "https://google.com`
+- `add link -n google -l https://google.com`
```
Comment (Devanshshah1309) [inline docs/UserGuide.md:441]: Add explanation for example.

File: docs/UserGuide.md [hunk 6]
```diff
@@ -276,7 +450,7 @@ Format:
 
 Examples:
 
-- `edit link 1 -n facebook -l "https://facebook.com`
+- `edit link 1 -n facebook -l https://facebook.com`
```
Comment (Devanshshah1309) [inline docs/UserGuide.md:453]: Add explanation for example.

File: docs/UserGuide.md [hunk 7]
```diff
@@ -364,19 +490,27 @@ Format:
...
 ### Saving the data
 
-TruthTable data are saved in the hard disk automatically after any command that changes the data. There is no need to
+TruthTable data is saved in the hard disk automatically after any command that changes the data. There is no need to
 save manually.
 
 ### Editing the data file
 
-TruthTable data are saved as a JSON file `[JAR file location]/data/truthtable.json`. Advanced users are welcome to
+TruthTable data is saved as a JSON file `[JAR file location]/data/addressbook.json`. Advanced users are welcome to
```
Comment (Devanshshah1309) [inline docs/UserGuide.md:508]: Is this intended? Can't we store it under `TruthTable.json` instead?

File: docs/UserGuide.md [hunk 8]
```diff
@@ -385,15 +519,15 @@ _Details coming soon ..._
 
 ---
 
-## FAQ
+# FAQ
```
Comment (Devanshshah1309) [inline docs/UserGuide.md:522]: Add more FAQs based on CS2101 feedback. Consider use-cases to come up with FAQs (and stalk other teams' UGs)

File: docs/UserGuide.md [hunk 9]
```diff
@@ -385,15 +519,15 @@ _Details coming soon ..._
...
 
 **Q**: How do I transfer my data to another Computer?<br>
 **A**: Install the app in the other computer and overwrite the empty data file it creates with the file that contains
 the data of your previous TruthTable home folder.
 
 ---
 
-## Command summary
+# Command summary
```
Comment (Devanshshah1309) [inline docs/UserGuide.md:530]: Yet to be included (we should probably shift this to the top and include links from this summary table to all other commands) -> pretty sick navigability bonus points then!

**Review by Devanshshah1309 — APPROVED** [review]

**Comment by marcuspang** [comment]
Remaining issues will be addressed in #84

File: docs/UserGuide.md [hunk 10]
```diff
@@ -131,6 +204,32 @@ Examples:
...
+**either** the words "Alex" or "Beatrice" will be returned. This is the same for emails.
+
+<div markdown="span" class="alert alert-info">
+:information_source: **Note:** The search is **not** case-sensitive.
+In other words, `find member -n alex` and `find member -n Alex` will return the same results.
+</div>
+
+<div markdown="span" class="alert alert-info">
+:information_source: **Note:** You can only find members using **either** emails or names. This means that you 
+should not use both `-n` and `-e` in the `find member` command. In essence, they are mutually exclusive.
```
[the remark this answers is not in the document]
Comment (marcuspang) [reply docs/UserGuide.md:220]: Yes either or

File: docs/UserGuide.md [hunk 11]
```diff
@@ -160,16 +275,19 @@ task list.
 Edits a specified task in the current team. The task index is compulsory, while the other fields are optional and
 will be overwritten accordingly.
 
+Multiple assignees can be assigned at one time, by specifying the indices of each of the members, separated by spaces.
+e.g. `edit task 1 -a 1 3 4` will assign members 1, 3 and 4.
+
```
[the remark this answers is not in the document]
Comment (marcuspang) [reply docs/UserGuide.md]: I dont get what you mean

File: docs/UserGuide.md [hunk 12]
```diff
@@ -223,36 +341,92 @@ Examples:
...
+Format:
+
+- `unmark INDEX`
+
+Examples:
+
+- `unmark 1` will mark the first task in the team as incomplete.
+
+### Find tasks : `find task`
+Find all tasks in the current team by name and displays them in the task list.
```
[the remark this answers is not in the document]
Comment (marcuspang) [reply docs/UserGuide.md:351]: Will be resolved in #84

File: docs/UserGuide.md [hunk 13]
```diff
@@ -223,36 +341,92 @@ Examples:
 
 - `delete task 1` will delete the first task of the current team.
 
+### Mark tasks as done: `mark`
+
+Mark a specified task as done. To undo this command, see the `unmark` command below.
```
[the remark this answers is not in the document]
Comment (marcuspang) [reply docs/UserGuide.md:332]: Will be resolved in #84

**Review by Devanshshah1309 — APPROVED** [review]
Preliminary changes LGTM. Approving since the rest of the changes will be made in a separate PR (as discussed internally)

**Pull request merged**